internal/cortex/provider.go

```diff
@@ -36,14 +36,14 @@ func Provider() *schema.Provider {
 			},
 			"api_key": {
 				Type:        schema.TypeString,
-				Optional:    true,
+				Required:    true,
 				Sensitive:   true,
 				DefaultFunc: schema.EnvDefaultFunc("CORTEX_API_KEY", ""),
 				Description: "API key, used as basic auth password.",
 			},
 			"tenant_id": {
 				Type:        schema.TypeString,
-				Optional:    true,
+				Required:    true,
```

Comment:
This is optional because it can be set at the resource level. This is so one provider can configure multiple tenants.

Reply:
There is no parameter to pass the api_key at resource level.[https://registry.terraform.io/providers/inuits/cortex/latest/docs/resources/rules](url)

Reply:
Indeed, now I remember why it is not mandatory. It is because this is not required to connect to a cortex cluster.

Reply:
ok makes sense then